oci/handler/directory.go

```diff
@@ -272,8 +275,15 @@ func (handle *DirectoryHandler) FetchImage(ctx context.Context, fullref, platfor
 	if err != nil {
 		return err
 	}
-
-	img, err := remote.Image(ref, remote.WithAuthFromKeychain(authn.DefaultKeychain))
+	img, err := remote.Image(ref,
+		remote.WithContext(ctx),
+		remote.WithPlatform(v1.Platform{
+			OS:           strings.Split(platform, "/")[0],
+			Architecture: strings.Split(platform, "/")[1],
+		}),
+		remote.WithUserAgent(version.UserAgent()),
+		remote.WithAuthFromKeychain(authn.DefaultKeychain),
```

Comment:
Would be nice to configure authentication, but leaving as-is for the general use case. We can re-address later.